examples/utils/example-utils/src/migrator/sameContainerMigrator.ts

```diff
@@ -0,0 +1,269 @@
+// Eventually it will be in charge of extracting the v1 data and calling migrationTool.finalizeMigration() with the transformed summary, but for now it's probably best to ignore it.
+export class SameContainerMigrator
+	extends TypedEventEmitter<ISameContainerMigratorEvents>
+	implements ISameContainerMigrator
+{
+	private _currentModel: ISameContainerMigratableModel;
+	public get currentModel(): ISameContainerMigratableModel {
+		return this._currentModel;
+	}
+
+	private _currentModelId: string;
+	public get currentModelId(): string {
+		return this._currentModelId;
+	}
+
+	public get migrationState(): SameContainerMigrationState {
+		return this._currentModel.migrationTool.migrationState;
+	}
+
+	public get connected(): boolean {
+		return this._currentModel.connected();
+	}
+
+	/**
+	 * If migration is in progress, the promise that will resolve when it completes.  Mutually exclusive with
+	 * _migratedLoadP promise.
+	 */
+	private _migrationP: Promise<void> | undefined;
+
+	/**
+	 * If loading the migrated container is in progress, the promise that will resolve when it completes.  Mutually
+	 * exclusive with _migrationP promise.
+	 */
+	private _migratedLoadP: Promise<void> | undefined;
+
+	/**
+	 * Detached model that is ready to attach. This is stored for retry scenarios.
+	 */
+	private _preparedDetachedModel: IDetachedModel<ISameContainerMigratableModel> | undefined;
+
+	public constructor(
+		private readonly modelLoader: IModelLoader<ISameContainerMigratableModel>,
+		initialMigratable: ISameContainerMigratableModel,
+		initialId: string,
+		private readonly dataTransformationCallback?: DataTransformationCallback,
+	) {
+		super();
+		this._currentModel = initialMigratable;
+		this._currentModelId = initialId;
+		this.takeAppropriateActionForCurrentMigratable();
+	}
+
+	/**
+	 * This method makes no assumptions about the state of the current migratable - this is particularly important
+	 * for the case that we just finished loading a migrated container, but that migrated container is also either
+	 * in the process of migrating or already migrated (and thus we need to load again).  It is not safe to assume
+	 * that a freshly-loaded migrated container is in collaborating state.
+	 */
+	private readonly takeAppropriateActionForCurrentMigratable = () => {
+		// TODO: Real stuff
+		// const migrationState = this._currentModel.migrationTool.migrationState;
+		// if (migrationState === "migrating") {
+		// 	this.ensureMigrating();
+		// } else if (migrationState === "migrated") {
+		// 	this.ensureLoading();
+		// } else {
+		// 	this._currentModel.migrationTool.once(
+		// 		"migrating",
+		// 		this.takeAppropriateActionForCurrentMigratable,
+		// 	);
+		// }
+		// TODO: One responsibility here will be generating the v2 summary
+
+		// TODO: Just logging these to quiet the "unused member" error.
+		console.log(this.ensureMigrating, this.ensureLoading);
```

Comment:
nit: might be nice to include this in the console log, since it confused me when I first tried out the example 😅 
```suggestion
		console.log("Just logging these to quiet the "unused member" error.", this.ensureMigrating, this.ensureLoading);
```